internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -460,6 +481,132 @@ func convertEnvSecurityGroupCfg(mft *manifest.Environment) (*template.SecurityGr
 	}, nil
 }
 
+func (s *LoadBalancedWebService) convertApplicationLoadBalancer() (*template.ALBListener, error) {
+	albConfig := s.manifest.RoutingRule
+	if albConfig.Disabled() || albConfig.IsEmpty() {
+		return nil, nil
+	}
+	var rules []template.ALBListenerRule
+	// build listener rule config from primary rule config from manifest.
+	rule, err := RoutingRuleConfigOrBoolConverter{
+		rule:         albConfig,
+		manifest:     s.manifest,
+		httpsEnabled: s.httpsEnabled,
+	}.convert()
+	if err != nil {
+		return nil, nil
+	}
+	aliasesFor, err := convertHostedZone(albConfig.Alias, albConfig.HostedZone)
+	if err != nil {
+		return nil, fmt.Errorf(`convert "http.alias" to string slice: %w`, err)
+	}
+	rules = append(rules, *rule)
```

Comment:
Maybe do this right after we get `rule`?

Reply:
In case if error we will be appending nil value of the `rule` to `rules`. I don't think that operation is even required in case of error.